src/vs/platform/extensionManagement/common/extensionManagementCLI.ts

```diff
@@ -172,6 +172,41 @@ export class ExtensionManagementCLI {
 		}
 	}
 
+	public async updateExtensions(profileLocation?: URI): Promise<void> {
+		const installedExtensions = await this.extensionManagementService.getInstalled(ExtensionType.User, profileLocation);
+
+		const installedExtensionsQuery: IExtensionInfo[] = [];
+		for (const extension of installedExtensions) {
+			if (!!extension.identifier.uuid) { // No need to check new version for an unpublished extension
+				installedExtensionsQuery.push({ ...extension.identifier, preRelease: extension.preRelease });
+			}
+		}
+
+		this.logger.trace(localize('updateExtensionsQuery', "Fetching latest versions for {0} extensions", installedExtensionsQuery.length));
+		const availableVersions = await this.extensionGalleryService.getExtensions(installedExtensionsQuery, { compatible: true }, CancellationToken.None);
+
+		const extensionsToUpdate: InstallExtensionInfo[] = [];
+		for (const newVersion of availableVersions) {
+			for (const oldVersion of installedExtensions) {
+				if (areSameExtensions(oldVersion.identifier, newVersion.identifier) && gt(newVersion.version, oldVersion.manifest.version)) {
+					extensionsToUpdate.push({
+						extension: newVersion,
+						options: { operation: InstallOperation.Update, installPreReleaseVersion: oldVersion.isPreReleaseVersion }
+					});
+				}
+			}
+		}
+
+		if (extensionsToUpdate.length) {
+			this.logger.info(localize('updateExtensionsNewVersionsAvailable', "Updating extensions: {0}", extensionsToUpdate.map(ext => ext.extension.identifier.id).join(', ')));
+			await this.extensionManagementService.installGalleryExtensions(extensionsToUpdate);
```

Comment:
Use the return result of `installGalleryExtensions` to log succuessfully updated and failed extensions

Reply:
Ok, now I understand better what you meant by "update the logic accordingly".
Last commit pushed should resolve that. Is it verbose in the right amount?

Successful update log example:
```
Updating extensions: eamodio.gitlens, tamasfe.even-better-toml
Successfully updated 2 extensions
```

Failed update log example:
```
Updating extensions: eamodio.gitlens, tamasfe.even-better-toml
Failed updating extensions: tamasfe.even-better-toml, eamodio.gitlens
```

Log for no update available:
```
No extension to update
```